agent/skyhook-agent/tests/test_controller.py

```
@@ -1104,6 +1104,26 @@ def test_interrupt_applies_all_commands(self, run_mock, datetime_mock):
             ])
 
+    @mock.patch("skyhook_agent.controller.do_interrupt")
+    def test_interrupt_bootstraps_copy_dir_before_running(self, do_interrupt_mock):
+        def _assert_copy_dir_exists(interrupt_data, root_mount, copy_dir):
+            self.assertTrue(os.path.exists(f"{root_mount}/{copy_dir}/config.json"))
+            return False
+
+        do_interrupt_mock.side_effect = _assert_copy_dir_exists
+
+        with self._setup_for_main() as (_, _, root_dir, copy_dir):
+            with set_env(SKYHOOK_RESOURCE_ID="scr-id-1_package_version"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)